wrappers/gin/gin.go

```diff
@@ -33,13 +88,17 @@ func wrapGinHandler(handler gin.HandlerFunc, hostname string, relativePath strin
 		}
 		c.Keys[TracerKey] = wrapperTracer
 		wrapper := epsagon.WrapGenericFunction(
-			handler,
-			config,
-			wrapperTracer,
-			false,
-			fmt.Sprintf("%s%s", hostname, relativePath),
+			handler, config, wrapperTracer, false, relativePath,
 		)
+		triggerEvent := createTriggerEvent(wrapperTracer, c, hostname)
+		wrapperTracer.AddEvent(triggerEvent)
 		wrapper.Call(c)
+		triggerEvent.Resource.Metadata["status_code"] = fmt.Sprint(c.Writer.Status())
```

Comment:
should this line (and the resource type set) be deffered? in case wrapper.Call panics

Reply:
If it is deffered then it might put a wrong status as no other was written (this is by looking at the implementation of gin..) So I will set it to 500 and this line will overwrite it only if the call did not panic.